connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceService.java

```diff
@@ -364,6 +366,16 @@ public CommonDeployment deploy(ServiceContainer serviceContainer) throws DeployE
 
                 DataSources dataSources = null;
                 if (dataSourceConfig != null) {
+                    String dsClsName = dataSourceConfig.getDataSourceClass();
+                    if (dsClsName != null) {
+                        Class<? extends DataSource> dsCls = driverClassLoader().loadClass(dsClsName).asSubclass(DataSource.class);
```

Comment:
@gaol what happens if the class does not implement the DataSource interface?

Reply:
If the class does not implement the DataSource interface, the data source creation fails as well, but with string `java.lang.ClassCastException` in the error message for the CLI user. I will try to improve the error message by catching this case. :) The same for the below comments :)